splash/qtrender_lua.py

```diff
@@ -328,7 +329,7 @@ def __call__(self, *args):
             args=escape_js(*args)
         )
         wrapper_script = get_process_errors_js(expr)
-        res = self.tab.evaljs(wrapper_script)
+        res = self.tab.evaljs(wrapper_script, result_protection=self.js_result_protection)
```

Comment:
I think both these options are fine. Adding it to sanitize may be a bit more convenient for users because they won't have to write `{ top: rect.x, right: rect.right, bottom: rect.bottom, left: rect.left }` themselves. A third option is to convert DOMRect to a special DOMRect Lua object - just add a few useful methods to `{ top=rect.x, right=rect.right, bottom=rect.bottom, left=rect.left }` table.